social_media_publishers/notification.py

```diff
@@ -0,0 +1,42 @@
+from datetime import datetime
+from social_media_publishers.publisher import Publisher
+from firestore.incidents import Incident
+from datetime import datetime
+from firestore.tokens import Token
+
+from exponent_server_sdk import (
+    DeviceNotRegisteredError,
+    PushClient,
+    PushMessage,
+    PushServerError,
+)
+
+
+class PushNotification(Publisher):
+    def __init__(self) -> None:
+        self.push_client = PushClient()
+
+    def publish(self, incident: Incident) -> datetime:
+        # get all tokens
+        tokens = Token.collection.fetch()
```

Comment:
we cannot load all tokens at once. Imagine there are 100K people registered, loading all in memory will cause out of memory error. we should load tokens in bathes.

the other concern is this is a HTTP request, which should be finished within certain time (1 minute). If there are a lot of tokens, it might take hours. You might need to shard the tokens (split into chunks (maybe using the first 1 or 2 characters of each token to group them, then process each group separately.).

The other way is to attach a incident Id or timestamp to each token. So for each token/user, we know which notification has been sent, so we just need to send any incidents newer than that timestamp next time.